File: csharp/MapleStory.OpenAPI/Src/Dto/CharacterPetEquipmentItemDTO.cs
using Newtonsoft.Json;

namespace MapleStory.OpenAPI.Dto
{
    /// <summary>
    /// 캐릭터 펫 장비 아이템 정보
    /// </summary>
    public class CharacterPetEquipmentItemDTO
    {
        /// <summary>
        /// 아이템 명
        /// </summary>
        [JsonProperty("item_name")]
        public string ItemName { get; set; }

        /// <summary>
        /// 아이템 아이콘
        /// </summary>
        [JsonProperty("item_icon")]
        public string ItemIcon { get; set; }

        /// <summary>
        /// 아이템 설명
        /// </summary>
        [JsonProperty("item_description")]
        public string ItemDescription { get; set; }

        /// <summary>
        /// 아이템 표기상 옵션
        /// </summary>
        [JsonProperty("item_option")]
        public List<CharacterPetEquipmentItemOptionDTO> ItemOption { get; set; }

        /// <summary>
        /// 업그레이드 횟수
        /// </summary>
        [JsonProperty("scroll_upgrade")]
        public long ScrollUpgrade { get; set; }

        /// <summary>
        /// 업그레이드 가능 횟수
        /// </summary>
        [JsonProperty("scroll_upgradeable")]
        public long ScrollUpgradeable { get; set; }
    }
}

File: java/src/main/java/dev/spiralmoon/maplestory/api/tms/dto/character/CharacterPetEquipmentItemDTO.java
package dev.spiralmoon.maplestory.api.tms.dto.character;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonSetter;
import com.fasterxml.jackson.annotation.Nulls;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.util.List;

/**
 * 寵物道具資訊
 */
@AllArgsConstructor
@NoArgsConstructor
@Data
@ToString
public class CharacterPetEquipmentItemDTO implements dev.spiralmoon.maplestory.api.common.dto.character.CharacterPetEquipmentItemDTO<CharacterPetEquipmentItemOptionDTO> {

    /**
     * 道具名稱
     */
    @JsonProperty("item_name")
    private String itemName;

    /**
     * 道具圖示
     */
    @JsonProperty("item_icon")
    private String itemIcon;

    /**
     * 道具描述
     */
    @JsonProperty("item_description")
    private String itemDescription;

    /**
     * 道具顯示選項
     */
    @JsonProperty("item_option")
    @JsonSetter(nulls = Nulls.AS_EMPTY)
    private List<CharacterPetEquipmentItemOptionDTO> itemOption;

    /**
     * 強化次數
     */
    @JsonProperty("scroll_upgrade")
    private int scrollUpgrade;

    /**
     * 道具可升級次數
     */
    @JsonProperty("scroll_upgradable")
    private int scrollUpgradable;

    /**
     * 道具外型
     */
    @JsonProperty("item_shape")
    private String itemShape;

    /**
     * 道具外型圖示
     */
    @JsonProperty("item_shape_icon")
    private String itemShapeIcon;
}
